sky/clouds/service_catalog/data_fetchers/fetch_lambda_cloud.py

```diff
@@ -0,0 +1,119 @@
+    'RTX6000': 24576,
+    'V100': 16384,
+}
+
+
+def name_to_gpu(name: str) -> str:
+    # Edge case
+    if name == 'gpu_8x_a100_80gb_sxm4':
+        return 'A100-80GB'
+    return name.split('_')[2].upper()
+
+
+def name_to_act(name: str) -> int:
+    return int(name.split('_')[1].replace('x', ''))
+
+
+def create_catalog(
+        api_key: str,  # pylint: disable=redefined-outer-name
+        output_path: str) -> None:
+    headers = {'Authorization': f'Bearer {api_key}'}
+    response = requests.get(ENDPOINT, headers=headers)
+    info = response.json()['data']
+
+    with open(output_path, mode='w') as f:  # pylint: disable=redefined-outer-name
+        writer = csv.writer(f, delimiter=',', quotechar='"')
+        writer.writerow([
+            'InstanceType', 'AcceleratorName', 'AcceleratorCount', 'vCPUs',
+            'MemoryGiB', 'Price', 'Region', 'GpuInfo', 'SpotPrice'
+        ])
+        # We parse info.keys() in reverse order so gpu_1x_a100_sxm4 comes before
+        # gpu_1x_a100 in the catalog (gpu_1x_a100_sxm4 has more availability).
+        for vm in reversed(list(info.keys())):
+            aname = name_to_gpu(vm)
+            act = name_to_act(vm) * 1.0
+            vcpus = info[vm]['instance_type']['specs']['vcpus'] * 1.0
+            mem = info[vm]['instance_type']['specs']['memory_gib'] * 1.0
+            price = info[vm]['instance_type']\
+                    ['price_cents_per_hour'] * 1.0 / 100
+            gpuinfo = {
+                'Gpus': [{
+                    'Name': aname,
+                    'Manufacturer': 'NVIDIA',
+                    'Count': act,
+                    'MemoryInfo': {
+                        'SizeInMiB': gpu_to_memory[aname]
+                    },
+                }],
+                'TotalGpuMemoryInMiB': gpu_to_memory[aname]
+            }
+            gpuinfo = json.dumps(gpuinfo).replace('"', "'")  # pylint: disable=invalid-string-quote
+            for r in regions:
+                writer.writerow(
+                    [vm, aname, act, vcpus, mem, price, r, gpuinfo, ''])
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--api-key', help='Lambda API key.')
+    parser.add_argument('--api-key-path',
+                        help='path of file containing Lambda API key.')
+    args = parser.parse_args()
+
+    api_key = args.api_key
+    if api_key is None:
+        if args.api_key_path is not None:
+            with open(args.api_key_path, mode='r') as f:
+                api_key = f.read().strip()
+        else:
+            # Read from ~/.lambda_cloud/lambda_keys
+            with open(LAMBDA_KEYS_PATH, mode='r') as f:
+                lines = [
+                    line.strip() for line in f.readlines() if ' = ' in line
+                ]
+                for line in lines:
+                    if line.split(' = ')[0] == 'api_key':
+                        api_key = line.split(' = ')[1]
+                        break
```

Comment:
Why don't we define a new function for this? Then we'll be able to remove the `# pylint: disable=redefined-outer-name` above.

Reply:
Done!